src/linked_list.py

```diff
@@ -50,17 +51,16 @@ def pop(self):
     def search(self, val):
         """Search through list for a node and return that node."""
         current = self.head
-        found = False
         if self.length > 0:
-            while current and found is False:
+            while current:
                 if current.value == val:
-                    found = True
-                    return current
+                    break
                 else:
                     current = current.next_node
-            if current is None:
+            if current:
+                return current.value
```

Comment:
Per the assignment specs, `search` should return a Node instance, not the value. Kind of weird, but that's what the specs say.